config/config.go

```diff
@@ -34,17 +34,18 @@ type ForwardTo struct {
 
 // ListenFrom configures how we listen for datapoints to forward
 type ListenFrom struct {
-	Type                       string
-	ListenAddr                 *string
-	MetricDeconstructor        *string
-	MetricDeconstructorOptions *string
-	Timeout                    *string
-	Name                       *string
-	ListenPath                 *string
-	JSONEngine                 *string
-	Encrypted                  *bool
-	TimeoutDuration            *time.Duration `json:"-"`
-	ServerAcceptDeadline       *time.Duration `json:"-"`
+	Type                           string
+	ListenAddr                     *string
+	MetricDeconstructor            *string
+	MetricDeconstructorOptions     *string
+	MetricDeconstructorOptionsJSON map[string]interface{}
```

Comment:
It may be useful to make a struct out of this rather than map[string]interface{} so we can validate it decodes correctly.